### DISCUSSION TITLE
Derived contexts for testing

### DISCUSSION THREAD

**Discussion by Aetet** [reatom/reatom#801] [Ideas]
#### What
It'll be great to have an opportunity to create derived contexts

#### Why
Create a partially populated context with data and have an opportunity to create a new environment quickly.
For example, create a new user, and profile, repopulate caches, etc.

Something like:
```ts
const ctx = createCtx();
const textAtom = atom('')

textAtom(ctx, 'newVal')

// Now it has prepopulated value for textAtom = 'newVal'
const prepopulatedContext = createDerived(ctx)
```

**Comment by Akiyamka** [comment]
This is a very interesting proposal. In the current version of reatom, it is difficult to implement such functionality. However, in the next major update (v4) this functionality will be implemented
